src/com/facebook/buck/ide/intellij/model/folders/AbstractIjSourceFolder.java

```diff
@@ -32,7 +32,12 @@
 
   public abstract boolean getIsTestSource();
```

Comment:
I suggest simplifying design and be consistent with java source folders and java test folders.

You can rename this method to `getIsTesFolder` and use it together with `getIsResourceFolder` to detect the folder type. You would need to add `JavaTestResourceFolder` class to be consistent with `TestFolder` and `SourceFolder`.